File: hdcd-r/tests/testthat/test-hdcd.R
test_that("hdcd_iris", {
    library(datasets)
    data(iris)
    X <- as.matrix(iris[, 1:4])

    expect(hdcd(X, "knn", "bs")$split_points() == c(50, 100))
    expect(hdcd(X, "change_in_mean", "bs")$split_points() == c(50, 100))
    expect(hdcd(X, "random_forest", "bs")$split_points() == c(50, 100))

    expect(hdcd(X, "knn", "sbs")$split_points() == c(50, 100))
    expect(hdcd(X, "change_in_mean", "sbs")$split_points() == c(50, 100))
    expect(hdcd(X, "random_forest", "sbs")$split_points() == c(50, 100))

    expect(hdcd(X, "knn", "wbs")$split_points() == c(50, 100))
    expect(hdcd(X, "change_in_mean", "wbs")$split_points() == c(50, 100))
    expect(hdcd(X, "random_forest", "wbs")$split_points() == c(50, 100))
})

File: hdcd-python/tests/test_hdcd.py
import numpy as np
import pytest
from hdcd import hdcd


@pytest.mark.parametrize("method", ["knn", "change_in_mean", "random_forest"])
@pytest.mark.parametrize("segmentation_type", ["sbs", "wbs", "bs"])
def test_hdcd(iris_dataset, method, segmentation_type):
    result = hdcd(iris_dataset, method, segmentation_type)
    np.testing.assert_array_equal(result.split_points(), [50, 100])
